php-custom-rules/src/main/java/org/sonar/samples/php/PHPRuleRepository.java

```diff
@@ -21,25 +21,23 @@
 
 
 import com.google.common.collect.ImmutableList;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Map;
-import org.sonar.plugins.php.api.visitors.PHPCustomRulesDefinition;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinitionAnnotationLoader;
+import org.sonar.plugins.php.api.visitors.PHPCustomRuleRepository;
 import org.sonar.samples.php.checks.ForbiddenFunctionUseCheck;
 import org.sonar.samples.php.checks.OtherForbiddenFunctionUseCheck;
-import org.sonar.squidbridge.annotations.AnnotationBasedRulesDefinition;
 
 /**
  * Extension point to define a PHP rule repository.
  */
-public class PHPRulesDefinition extends PHPCustomRulesDefinition {
-
-  /**
-   * Provide the repository name
-   */
-  @Override
-  public String repositoryName() {
-    return "MyCompany Custom Repository";
-  }
+public class PHPRuleRepository implements RulesDefinition, PHPCustomRuleRepository {
```

Comment:
@pynicolas it's a bit confusing to have `repository` implementing `RulesDefinition` interface

Reply:
Could you please elaborate?

Reply:
@pynicolas how about something generic for this class name, e.g. `PHPCustomRules` or `MyPhpRules`?